codebuild/bin/run_cppcheck.sh

```diff
@@ -16,21 +16,27 @@
 set -e
 
 usage() {
-    echo "run_cppcheck.sh install_dir"
+    echo "run_cppcheck.sh install_dir cache_dir"
     exit 1
 }
 
-if [ "$#" -ne "1" ]; then
+if [ "$#" -ne "2" ]; then
     usage
 fi
 
 INSTALL_DIR=$1
+CACHE_DIR=$2
+mkdir -p "$CACHE_DIR" || true
 
 CPPCHECK_EXECUTABLE=${INSTALL_DIR}/cppcheck
 
 FAILED=0
 $CPPCHECK_EXECUTABLE --version
-$CPPCHECK_EXECUTABLE --std=c99 --error-exitcode=-1 --quiet --force -j 8 --enable=all --template='[{file}:{line}]: ({severity}:{id}) {message}' --inline-suppr --suppressions-list=codebuild/bin/cppcheck_suppressions.txt -I . -I ./tests api bin crypto error stuffer ./tests/unit tls utils || FAILED=1
+
+# NOTE: cppcheck should be run in single thread to ensure we are check for `unusedFunction`. Do not add the `-j` flag.
+$CPPCHECK_EXECUTABLE --std=c99 --error-exitcode=-1 --force --enable=all -j 2 --template='[{file}:{line}]: ({severity}:{id}) {message}' --inline-suppr --cppcheck-build-dir="$CACHE_DIR" --suppressions-list=codebuild/bin/cppcheck_suppressions.txt -I . -I api crypto || FAILED=1
```

Comment:
replace `-j 2` with `-j $(nproc)`

Reply:
> # NOTE: cppcheck should be run in single thread to ensure we are check for `unusedFunction`. Do not add the `-j` flag.
so using `-j 2` or more actually disables the unusedFunction check. So i dont think we wan to specify that.

Reply:
[Docs](https://cppcheck.sourceforge.io/manual.pdf) are suggesting this works with cppcheck-build-dir  which appears to be false. 
```
Whole program analysis does not work when multiple threads are used; unless
you use a cppcheck build dir. For instance, the unusedFunction warnings require
whole program analysis.
```

 Is the longer runtime worth this check?